app/models/race_entry.rb

```diff
@@ -3,21 +3,18 @@
 class RaceEntry < ActiveRecord::Base
   belongs_to :racer
   belongs_to :race_edition
-     
+
   validates :racer, presence: true
   validates :race_edition, presence: true
-
+  validates_uniqueness_of :racer_id, scope: :race_edition_id,
+                          message: 'may be added to a race_edition only once'
 
   def elapsed_time
-    if time? then
-      ChronicDuration.output(time)
-    else
-      'NA'
-    end
+    time && ChronicDuration.output(time) || 'NA'
```

Comment:
In Ruby everything evaluates to truthy or falsey, allowing shortcuts like this. if `time` is nil, then we pass through the `||` operator and return "NA".